lib/opencensus_phoenix/instrumenter.ex

```diff
@@ -87,12 +90,32 @@ defmodule OpencensusPhoenix.Instrumenter do
     :ocp.with_span_ctx(parent_span_ctx)
   end
 
-  def span_name(conn) do
+  defp span_name(_conn, %{route: route}) do
+    route
+  end
+
+  defp span_name(conn, _) do
+    controller_action(conn)
+  end
+
+  defp controller_action(conn) do
     controller = Phoenix.Controller.controller_module(conn)
     action = Phoenix.Controller.action_name(conn)
     "#{controller}.#{action}"
   end
 
+  defp route_info(%{method: method, request_path: request_path, host: host} = conn) do
+    route_info_present? = :erlang.function_exported(Phoenix.Router, :route_info, 4)
```

Comment:
This seems like a good use case for a macro since Elixir, I think, makes it easy to check the version (or maybe even check if the function exists during compilation?). Should be able to just compile the function to use the new route info function if it is being built in a project with a new enough Phoenix in my opinion.

Reply:
Hmm....is the phoenix dependency guaranteed to be compiled before us if we reference one of its module names? I've had problems before looking for modules/functions defined at compilation time due to compilation order changing in the future.

Reply:
Actually, yeah a combination of `Code.ensure_compiled/1` and then after that checking for function definition should work. Other libs do the same.

Reply:
How does it look now? I Just conditionally defined each function.

Reply:
Looks good. Just had a question about apply and then can merge this.